test/aggregates/aggregates_test.go

```diff
@@ -0,0 +1,208 @@
+// +build integration
+
+/*
+Copyright 2018 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+package aggregates
+
+import (
+	"fmt"
+	"github.com/v3io/v3io-tsdb/pkg/tsdb"
+	"github.com/v3io/v3io-tsdb/pkg/tsdb/tsdbtest"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"testing"
+	"time"
+)
+
+var basetime int64
+
+type TestConfig struct {
+	Interval     int64
+	TimeVariance int
+	TestDuration string
+	NumMetrics   int
+	NumLabels    int
+	Values       []float64
+	QueryFunc    string
+	QueryStart   string
+	QueryStepSec int64
+}
+
+type metricContext struct {
+	lset utils.Labels
+	ref  uint64
+}
+
+func TestAggregates(t *testing.T) {
+
+	testConfig := TestConfig{
```

Comment:
I would recommend separating the test onto stages:
1. Setup - prepare the data and prerequisites required for running the test like table with desired partitioning, etc
2. TearDown - cleanup after test (could be skipped if test has failed to make investigation possible)
3. The test code itself.

Nice example: `pkg/aggregate/aggregate_test.go`

Reply:
This integration test wont pass twice.